docs/source/development/defining_new_models.md

```diff
@@ -0,0 +1,216 @@
+be provided when class inheritance occurs, so that the model can automatically be added
+to the registry under that name. N.B. The `self.name` attribute for the class is
+automatically changed to this name.
+
+```python
+class FreshWaterModel(BaseModel, model_name="freshwater"):
+    """Docstring describing model.
+
+    Args:
+        Describe arguments here
+    """
+```
+
+With the basic class now defined an `__init__` function should be added to the class.
+This should do a few things. Firstly, it should check that the provided initialisation
+values are sane (e.g. number of ponds is not negative). Secondly, any model specific
+attributes (e.g. number of ponds) should be stored, and general attributes should be
+passed to the `__init__` of the base class. Finally, names of model specific attributes
+should be appended to `__repr` so that they can be found (and printed) by `BaseModel`'s
+`__repr__` function.
+
+```python
+def __init__(
+    self,
+    update_interval: timedelta64,
+    start_time: datetime64,
+    no_of_ponds: int,
+    **kwargs: Any,
+):
+        
+    # Sanity checking of input variables goes here
+    if no_of_ponds < 0:
+        log_and_raise(
+        "There has to be at least one pond in the freshwater model!",
+            InitialisationError,
+        )
+        
+    # Provide general attributes to the __init__ of the base class
+    super().__init__(update_interval, start_time, **kwargs)
+    # Store model specific details as attributes.
+    self.no_of_ponds = int(no_of_ponds)
+    # Save attribute names to be used by the __repr__
+    self._repr.append("no_of_ponds")
+```
+
+Generally, `__init__` functions should only handle the most basic sanity checking and
+should take in fairly simple variables. However, our configuration is stored in a
+complex configuration object (a dictionary at the moment), which needs to be unpacked.
+Furthermore, some of the information contained in the configuration has to be converted
+to a more usable form (particularly time/date information). To accomplish this a
+`from_config` factory method has to be defined. This function unpacks the configuration,
+performs more complex sanity checks, and if necessary converts the information from the
+configuration to a more usable form. This is a class method, and so creates a
+initialised instance of the `Model` (and if it can't is raises an error).
+
+```python
+@classmethod
+def from_config(cls, config: dict[str, Any]) -> FreshWaterModel:
+    """Factory function to initialise the freshwater model from configuration.
+
+    This function unpacks the relevant information from the configuration file, and
+    then uses it to initialise the model. If any information from the config is
+    invalid rather than returning an initialised model instance an error is raised.
+
+    Args:
+        config: The complete (and validated) virtual rainforest configuration.
```

Comment:
General comment/question: do we capitalize 'Virtual Rainforest' or 'virtual rainforest'

Reply:
I think "Virtual Rainforest" in text? @robewers01

Reply:
I've been capitalising it generally (though I didn't here 😅), but agree that should probably settle on an agreed style